### PR TITLE
Week2 petergreve

### PR DESCRIPTION
**Pull request by petergreve**

### CODE REVIEW

File: petergreve/week2/Ass3/index.html [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+			return 	<div style={style}>
+						{this.props.errorMessage}
+					</div>
+		}
+
+	});
+
+	var App = React.createClass({
+		getInitialState: function() {
+		    return {
```
Comment (hulvei3) [inline petergreve/week2/Ass3/index.html]: Good idea to show me your data-model!

But of cause nothing you would have in production, I hope :)

File: petergreve/week2/Ass3/index.html [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+					    					key={i}
+					    					handleInput={that.handleInput}
+					    					archiveMessage={that.archiveMessage}
+					    					createMessage={that.createMessage}
+					    					deleteMessage={that.deleteMessage}
+					    					messages={this.filterMessages(list.listId)}
+					    					archived={this.filterArchived(list.listId)}
+					    					moveMessage={that.moveMessage}
+					    					list={list} />;
+					   		},this)}
```
Comment (hulvei3) [inline petergreve/week2/Ass3/index.html]: You are both binding the map function with `this` and using a this in another variable. Do one or the other

File: petergreve/week2/Ass3/index.html [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+					    </ol>			
+					</div>
+		},
+		handleInput: function(message) {
```
Comment (hulvei3) [inline petergreve/week2/Ass3/index.html]: for the future: it's a bit annoying to read code with this kind of indentation :)

File: petergreve/week2/Ass3/index.html [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+		}
+
+	});
+
+	var App = React.createClass({
+		getInitialState: function() {
+		    return {
+		    		listCount: 2,
+		    		lists: [{listId: 0, listName: "first list"},{listId: 1, listName: "second list"}],
+		    		messageCount: 4,
```
Comment (hulvei3) [inline petergreve/week2/Ass3/index.html]: things like `listCount` and `messageCount` doesn't really feed any purpose when you have access to the data-source itself. Now you have to maintain them every time you update the lists and messages.

File: petergreve/week2/Ass3/index.html [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+		},
+		createList: function() {
+			var newList = [{
+				listId: this.state.listCount,
+				listName: this.newListName.value
+			}];
+
+			this.setState({
+		          lists: this.state.lists.concat(newList),
+		          listCount: this.state.listCount + 1
```
Comment (hulvei3) [inline petergreve/week2/Ass3/index.html]: see my comment above regarding the `-count` vars

File: petergreve/week2/Ass3/index.html [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+		      	});
+		},
+		createMessage: function(text, listId) {
+			if (text.length<200) {
```
Comment (hulvei3) [inline petergreve/week2/Ass3/index.html]: I have seen this validation twice now?

File: petergreve/week2/Ass3/index.html [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+									{messageId: 2 ,listId: 2, text: "testmessage for list 2", isArchived: false}
+		    					 ]
+		    	};
+		},
+		render: function() {
+			var that = this;
+			return 	<div>
+		    			<input type="text" ref={(component) => this.newListName = component} />
+						<button type="button" onClick={this.createList}>New List</button>
+						<ol>
```
Comment (hulvei3) [inline petergreve/week2/Ass3/index.html]: I'm just curious: why an ordered-list?

File: petergreve/week2/Ass3/index.html (hunk 3 above)
Comment (hulvei3) [inline petergreve/week2/Ass3/index.html]: and this function doesn't seem to be used, hence your error-message is not working.

File: petergreve/week2/Ass3/index.html [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+			          messageCount: this.state.messageCount + 1
+			      	});
+			} 
+		},
+		filterMessages: function(listId) {
+			return this.state.allMessages.filter((m) => {
+				return m.isArchived == false && m.listId == listId;
+			});
+		},
+		filterArchived: function(listId) {
```
Comment (hulvei3) [inline petergreve/week2/Ass3/index.html]: Nice to see the use of filters.

File: petergreve/week2/Ass3/index.html [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+
+				this.setState({
+			          allMessages: this.state.allMessages.concat(newMessage),
+			          messageCount: this.state.messageCount + 1
+			      	});
+			} 
+		},
+		filterMessages: function(listId) {
+			return this.state.allMessages.filter((m) => {
+				return m.isArchived == false && m.listId == listId;
```
Comment (hulvei3) [inline petergreve/week2/Ass3/index.html]: it depends on religion, but you could write `!m.isArchived`, but it do seem more readable this way

File: petergreve/week2/Ass3/index.html [hunk 10]
```diff
@@ -0,0 +1,233 @@
...
+			var newAllMessages = this.state.allMessages.filter((m) => {
+				return m.messageId != messageId;
+			});
+
+			this.setState({
+				allMessages: newAllMessages
+			});
+		},
+		archiveMessage: function(messageId, isArchived) {
+			var archivedMessage = this.state.allMessages.filter((m) => {
```
Comment (hulvei3) [inline petergreve/week2/Ass3/index.html]: this is where we would use `.find()` normally, but this need polyfills in IE (we do that in e-conomic)

File: petergreve/week2/Ass3/index.html [hunk 11]
```diff
@@ -0,0 +1,233 @@
...
+		deleteMessage: function(messageId) {
+			var newAllMessages = this.state.allMessages.filter((m) => {
+				return m.messageId != messageId;
+			});
+
+			this.setState({
+				allMessages: newAllMessages
+			});
+		},
+		archiveMessage: function(messageId, isArchived) {
```
Comment (hulvei3) [inline petergreve/week2/Ass3/index.html]: this seems more to _toggle_ the a message's archive than just archive it. Maybe a name change?

File: petergreve/week2/Ass3/index.html [hunk 12]
```diff
@@ -0,0 +1,233 @@
...
+			var newAllMessages = this.state.allMessages.filter((m) => {
+				return m.messageId != messageId;
+			});
+
+			this.setState({
+				allMessages: newAllMessages.concat([archivedMessage])
+			});
+		},
+		moveMessage: function(messageId, down) {
+			var archivedMessage = this.state.allMessages.filter((m) => {
```
Comment (hulvei3) [inline petergreve/week2/Ass3/index.html]: the naming `archivedMessage` is really confusing here :)

File: petergreve/week2/Ass3/index.html [hunk 13]
```diff
@@ -0,0 +1,233 @@
...
+			else if (!down && archivedMessage.listId > 0)
+				archivedMessage.listId = archivedMessage.listId - 1;
+			else return
+
+			var newAllMessages = this.state.allMessages.filter((m) => {
+				return m.messageId != messageId;
+			});
+
+			this.setState({
+				allMessages: newAllMessages.concat([archivedMessage])
```
Comment (hulvei3) [inline petergreve/week2/Ass3/index.html]: I get why you are doing this, but you are in fact still manipulating state when you set the `listId`. But we will talk about that today

File: petergreve/week2/Ass3/index.html [hunk 14]
```diff
@@ -0,0 +1,233 @@
...
+</head>
+<body>
+
+<div id="root"> </div>
+	<script type="text/babel">
+
+	var List = React.createClass({
+
+		handleInput: function() {
+		  	var message = this.input.value;
```
Comment (hulvei3) [inline petergreve/week2/Ass3/index.html]: it seems to me that this is a good place to handle validation. Then you don't get into the creation mode at all, if the message is not valid

File: petergreve/week2/Ass3/index.html [hunk 15]
```diff
@@ -0,0 +1,233 @@
...
+					     					<button type="button" onClick={that.handleUnarchiveClick.bind(null, message)}>Unarchive</button>
+					     				</div>;
+					     	})}
+					    </ol>
+      				</div>
+		}
+
+	});
+
+	var ErrorMessage = React.createClass({
```
Comment (hulvei3) [inline petergreve/week2/Ass3/index.html]: you don't use this?

File: petergreve/week2/Ass3/index.html (hunk 1 above)
Comment (petergreve) [inline petergreve/week2/Ass3/index.html]: Yes, it is just for testing purposes and to get a quick view of the datamodel

File: petergreve/week2/Ass3/index.html (hunk 2 above)
Comment (petergreve) [inline petergreve/week2/Ass3/index.html]: You are right, I will remove the this=that thing

File: petergreve/week2/Ass3/index.html (hunk 3 above)
Comment (petergreve) [inline petergreve/week2/Ass3/index.html]: Indentation should be fixed

File: petergreve/week2/Ass3/index.html (hunk 4 above)
Comment (petergreve) [inline petergreve/week2/Ass3/index.html]: You are so right

File: petergreve/week2/Ass3/index.html (hunk 6 above)
Comment (petergreve) [inline petergreve/week2/Ass3/index.html]: handleInput is redundant legacy code and can be deleted

File: petergreve/week2/Ass3/index.html (hunk 7 above)
Comment (petergreve) [inline petergreve/week2/Ass3/index.html]: Just to add some very minimal ux - if you can call it that

File: petergreve/week2/Ass3/index.html (hunk 12 above)
Comment (petergreve) [inline petergreve/week2/Ass3/index.html]: I can follow that - I've changed the names

**Comment by hulvei3** [comment]
Will you be implementing any changes regarding the comments, or are you waiting for me to merge?

**Comment by petergreve** [comment]
I will implement some changes today...

**Pull request merged**